debian/arduino-app-cli/usr/sbin/arduino-avahi-serial.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+# Configure Avahi with the serial number.
+
+
+TARGET_FILE="/etc/avahi/services/arduino.service"
+MARKER_LINE="</service>"
+SERIAL_NUMBER_PATH="/sys/devices/soc0/serial_number"
+
+echo "Configuring Avahi with serial number for network discovery..."
+
+if [ ! -f "$SERIAL_NUMBER_PATH" ]; then
+    echo "Error: Serial number path not found at $SERIAL_NUMBER_PATH." >&2
+    exit 1 
+fi
+
+
+if [ ! -w "$TARGET_FILE" ]; then
+    echo "Error: Target file $TARGET_FILE not found or not writable." >&2
+    exit 1
+fi
+
+SERIAL_NUMBER=$(cat "$SERIAL_NUMBER_PATH")
+
+if [ -z "$SERIAL_NUMBER" ]; then
+    echo "Error: Serial number file is empty." >&2
+    exit 1 
+fi
+
+if grep -q "serial_number=${SERIAL_NUMBER}" "$TARGET_FILE"; then
```

Comment:
I would do the check only on the `serial_number` properties to be sure we don't include multiple times the property `serial_number`